File: matlab/lib/+Ice/SliceInfo.m
classdef (Sealed) SliceInfo < handle
    %SLICEINFO Encapsulates the details of a class slice with an unknown type.
    %
    %   SliceInfo Properties:
    %     typeId - The Slice type ID for this slice.
    %     compactId - The Slice compact type ID for this slice.
    %     bytes - The encoded bytes for this slice, including the leading size integer.
    %     hasOptionalMembers - Whether or not the slice contains optional members.
    %     isLastSlice - Whether or not this is the last slice.
    %     instances - The class instances referenced by this slice.

    % Copyright (c) ZeroC, Inc.

    properties (SetAccess = immutable)
        %TYPEID The Slice type ID for this slice.
        %   character vector
        typeId (1, :) char

        %COMPACTID The Slice compact type ID for this slice.
        %   int32 scalar
        compactId (1, 1) int32

        %BYTES The encoded bytes for this slice, including the leading size integer.
        %   uint8 vector
        bytes (1, :) uint8

        %HASOPTIONALMEMBERS Whether or not the slice contains optional members.
        %   logical scalar
        hasOptionalMembers (1, 1) logical

        %ISLASTSLICE Whether or not this is the last slice.
        %   logical scalar
        isLastSlice (1, 1) logical
    end
    properties
        %INSTANCES The class instances referenced by this slice.
        %   cell array of Ice.Value
        instances (1, :) cell = {}
    end
    methods (Hidden)
        function obj = SliceInfo(typeId, compactId, bytes, hasOptionalMembers, isLastSlice)
            obj.typeId = typeId;
            obj.compactId = compactId;
            obj.bytes = bytes;
            obj.hasOptionalMembers = hasOptionalMembers;
            obj.isLastSlice = isLastSlice;
        end
    end
end

File: java/src/Ice/src/main/java/com/zeroc/Ice/SliceInfo.java
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

package com.zeroc.Ice;

/**
 * SliceInfo encapsulates the details of a slice for an unknown class or exception type.
 **/
public class SliceInfo
{
    /**
     * The Slice type ID for this slice.
     **/
    public String typeId;

    /**
     * The Slice compact type ID for this slice.
     **/
    public int compactId;

    /**
     * The encoded bytes for this slice, including the leading size integer.
     **/
    public byte[] bytes;

    /**
     * The class instances referenced by this slice.
     **/
    public com.zeroc.Ice.Value[] instances;

    /**
     * Whether or not the slice contains optional members.
     **/
    public boolean hasOptionalMembers;

    /**
     * Whether or not this is the last slice.
     **/
    public boolean isLastSlice;
}
